numpy/core/_methods.py

```diff
@@ -140,13 +141,18 @@ def _clip(a, min=None, max=None, out=None, *, casting=None, **kwargs):
         return _clip_dep_invoke_with_casting(
             um.clip, a, min, max, out=out, casting=casting, **kwargs)
 
-def _mean(a, axis=None, dtype=None, out=None, keepdims=False):
+def _mean(a, axis=None, dtype=None, out=None, keepdims=False, where=True):
     arr = asanyarray(a)
 
     is_float16_result = False
-    rcount = _count_reduce_items(arr, axis)
-    # Make this warning show up first
-    if rcount == 0:
+
+    if isinstance(where, bool):
+        rcount = _count_reduce_items(arr, axis)
+    else:
+        logic_mask = where * nu.ones_like(arr, dtype=nt.bool_)
```

Comment:
This would be faster as `np.broadcast_to(where, arr.shape)`

Reply:
Alright! Is it OK if I then add `import numpy as np` at the top of the file? I am just wondering because all other imports are very specifiy. I tried importing just `broadcast_to` from `numpy.lib.stride_tricks` but this leads to fails in imports (circular imports?).

Reply:
You could import from stride_tricks locally within this function, which would break the cycle.

Reply:
Alright thank you for the hint! By now I need also `np.maximum` and `np.any`, so I would keep it as is.

Reply:
All those other functions are accessible under `nu` or some other import in this file.

Reply:
:+1:  Took me a while to figure out where which function comes from, but now `np.maximum` is replaced with `um.maximum` and `np.any` with `umr_any` (keeping in mind the differing default argument for axis).